src/applications/caregivers/components/PreSubmitInfo/SignatureInput.jsx

```diff
@@ -38,47 +40,41 @@ const SignatureInput = props => {
    * signatureMatches: compare the normalizedSignature string again the desired value
    * isSignatureComplete: complete if the value & desired string match and the parent checkbox is checked
    */
-  const normalizedSignature = signature.value.replace(/ +(?= )/g, '');
-
+  const normalizedSignature = data.value.replace(/ +(?= )/g, '');
   const signatureMatches =
     normalizedSignature.toLocaleLowerCase() === fullName.toLocaleLowerCase();
-
   const isSignatureComplete = signatureMatches && isChecked;
 
-  // set blur event for the input
-  const onBlur = useCallback(
+  const handleChange = value => {
+    setSignatures(prevState => ({ ...prevState, [label]: value }));
```

Comment:
Noting that I cannot get test coverage on the nested function to grab the `prevState` of this value. If anyone has any experience successfully stubbing a React `useState` with prevState, I'd love to learn how.

Reply:
Hmm. I think this may be a code smell. It feels weird that this component needs a setter to update multiple signatures, when really all it should care about is the one signature. I wonder if instead of passing the `setSignatures` method we should instead define a callback method in the parent component that encapsulates updating the prevState? That way we would just be testing that the callback gets called in this component, since we shouldn't really have to worry about the state of the parent component here like that? Example callback to replace passing `setSignatures`: 
```
  const updateSignatures = (label, value) => {
    setSignatures(prevState => ({ ...prevState, [label]: value }));
  }
  ```
  
I also noticed we are passing the `signatures` object into the signature checkbox components and we aren't doing anything with them there?